### PR TITLE
Core/Criterias: Fix an old criteria condition and added some new ones

### PR DESCRIPTION
**Pull request by Krudor**
**Changes proposed:**

-  Redefine and fix an old criteria condition
-  Name and research on some previously unknown criteria conditions

**Target branch(es):** master

**Issues addressed:** Fixes a bunch of boss achievements that require a certain map difficulty

**Tests performed:** Builds and in-game tested

### CODE REVIEW

File: src/server/game/DataStores/DBCEnums.h [hunk 1]
```diff
@@ -217,16 +217,33 @@ enum CriteriaAdditionalCondition
...
+    //CRITERIA_ADDITIONAL_CONDITION_UNK86                         = 86, // Some external event id
+    //CRITERIA_ADDITIONAL_CONDITION_UNK87                         = 87, // Achievement id
     CRITERIA_ADDITIONAL_CONDITION_BATTLE_PET_SPECIES            = 91,
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_FOLLOWER_ENTRY       = 144,
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_FOLLOWER_QUALITY     = 145,
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_FOLLOWER_LEVEL       = 146,
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_RARE_MISSION         = 147, // NYI
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_BUILDING_LEVEL       = 149, // NYI
     CRITERIA_ADDITIONAL_CONDITION_GARRISON_MISSION_TYPE         = 167, // NYI
+    CRITERIA_ADDITIONAL_CONDITION_PLAYER_ITEMLEVEL              = 169, // NYI
```
Comment (Shauren) [inline src/server/game/DataStores/DBCEnums.h]: make that *_ITEM_LEVEL

File: src/server/game/Achievements/CriteriaHandler.cpp [hunk 2]
```diff
@@ -1621,11 +1621,14 @@ bool CriteriaHandler::AdditionalRequirementsSatisfied(ModifierTreeNode const* tr
...
             break;
         }
-        case CRITERIA_ADDITIONAL_CONDITION_MAP_DIFFICULTY: // 20
-            if (uint32(referencePlayer->GetMap()->GetDifficultyID()) != reqValue)
-                return false;
+        case CRITERIA_ADDITIONAL_CONDITION_MAP_DIFFICULTY_OLD: // 20
+        {
+            DifficultyEntry const* difficulty = sDifficultyStore.LookupEntry(referencePlayer->GetMap()->GetDifficultyID());
+            if (!difficulty || difficulty->OldEnumValue != reqValue)
+                    return false;
```
Comment (Shauren) [inline src/server/game/Achievements/CriteriaHandler.cpp]: fail indent

Comment (Krudor) [inline src/server/game/Achievements/CriteriaHandler.cpp]: Fixed, forgot to sync.

**Comment by Aokromes** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Achievements/CriteriaHandler.cpp:1627:57: fatal error: 
      comparison of integers of different signs: 'const int8'
      (aka 'const signed char') and 'uint32' (aka 'unsigned int')
      [-Wsign-compare]
            if (!difficulty || difficulty->OldEnumValue != reqValue)
                               ~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~

**Pull request closed without merging**